pennylane/transforms/zx/converter.py

```diff
@@ -0,0 +1,302 @@
+        vertex = graph.add_vertex(VertexType.BOUNDARY, i, 0)
+        inputs.append(vertex)
+        q_mapper.set_prev_vertex(i, vertex)
+        q_mapper.set_next_row(i, 1)
+        q_mapper.set_qubit(i, i)
+
+    # Expand the tape to be compatible with PyZX and add rotations first for measurements
+    stop_crit = qml.BooleanFn(lambda obj: obj.name in gate_types)
+    tape = qml.tape.tape.expand_tape(tape, depth=10, stop_at=stop_crit, expand_measurements=True)
+
+    expanded_tape = []
+
+    # Define specific decompositions
+    for op in tape.operations:
+        if op.name == "PauliY":
+            decomp = [
+                qml.S(wires=0),
+                qml.RX(np.pi / 2, wires=0),
+                qml.RZ(np.pi + np.pi, wires=0),
+                qml.RX(np.pi / 2, wires=0),
+                qml.RZ(3 * np.pi, wires=0),
+                qml.S(wires=0),
+            ]
+            expanded_tape.extend(decomp)
+        elif op.name == "RY":
+            theta = op.data[0]
+            decomp = [
+                qml.RX(np.pi / 2, wires=0),
+                qml.RZ(theta + np.pi, wires=0),
+                qml.RX(np.pi / 2, wires=0),
+                qml.RZ(3 * np.pi, wires=0),
+            ]
+            expanded_tape.extend(decomp)
+        else:
+            expanded_tape.append(op)
+
+    expanded_tape = qml.tape.QuantumTape(expanded_tape, tape.measurements, [])
+
+    # Create graph from circuit in the tape (operations, measurements)
+    for op in expanded_tape.operations:
+
+        # Check that the gate is compatible with PyZX
+        name = op.name
+        if name not in gate_types:
+            raise qml.QuantumFunctionError(
+                "The expansion of the tape failed, PyZX does not support", name
+            )
+
+        # Apply wires and parameters
+        map_gate = gate_types[name]
+
+        par = [param / np.pi for param in op.parameters]
+        wires = list(op.wires)
+
+        # Max number of parameter is one
+        if par:
+            args = []
+            args.extend(wires)
+            args.extend(par)
+        else:
+            args = wires
+
+        gate = map_gate(*args)
+        gate.to_graph(graph, q_mapper, c_mapper)
+
+    row = max(q_mapper.max_row(), c_mapper.max_row())
+
+    outputs = []
+    for mapper in (q_mapper, c_mapper):
+        for label in mapper.labels():
+            qubit = mapper.to_qubit(label)
+            vertex = graph.add_vertex(VertexType.BOUNDARY, qubit, row)
+            outputs.append(vertex)
+            pre_vertex = mapper.prev_vertex(label)
+            graph.add_edge(graph.edge(pre_vertex, vertex))
+
+    graph.set_inputs(tuple(inputs))
+    graph.set_outputs(tuple(outputs))
+
+    return graph
+
+
+def from_zx(graph, split_phases=True):
+    """It converts a graph from PyZX to a PennyLane tape, if graph is diagram-like.
+    Args:
+        graph(Graph): ZX graph in PyZX.
+        split_phases(bool): If True the phases are split.
```

Comment:
Might be useful to have a definition of what "phase are split" means.

Reply:
Yes I will add examples here

Reply:
Yes more documentation is coming.